tests/proxytest.py

```diff
@@ -424,14 +438,14 @@ async def test_websockets(proxy, launch_backend):
     launch_backend(default_backend_port, "ws")
 
     await exponential_backoff(
-        utils.check_host_up, "Traefik not reacheable", ip="localhost", port=traefik_port
+        utils.check_host_up, "Traefik not reacheable", ip="127.0.0.1", port=traefik_port
```

Comment:
But why?

Reply:
`localhost` was resolving to `::1` in my test container. I'm running everything in docker containers, where IPv6 is disabled by default, so I don't think `::1` is a usable address. I was experiencing a load of 'cannot bind ::1 address' test error messages, which I hoped this would fix. Those errors might have been caused by some other problem, I admit. I haven't specifically reverted this change and re-tested after everything else works tbh.

Reply:
> localhost was resolving to ::1 in my test container

It looks like something is wrong with your Docker installation. For example, you have IPv6 disabled on the kernel level (`/proc/sys/net/ipv6/conf/all/disable_ipv6` is set to 1), but there is no `ipv6: false` in your `/etc/docker/daemon.json` file.

Reply:
I thought IPv6 was just disabled by default with docker: https://docs.docker.com/config/daemon/ipv6/

On my current system:-
```
$ cat /proc/sys/net/ipv6/conf/all/disable_ipv6
0
$ ls /etc/docker/
key.json
$
```

However, also (while in the container):-
```
root@8933dcac4db4:/srv/jupyterhub# cat /etc/hosts
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
172.23.0.4      8933dcac4db4
root@8933dcac4db4:/srv/jupyterhub# openssl s_client -connect ip6-localhost:8081
CONNECTED(00000003)
^C
root@8933dcac4db4:/srv/jupyterhub#
```

So yes, changing `localhost` to `127.0.0.1` was probably not what fixed whatever issue I was facing at the time!